keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -250,10 +251,23 @@ limitations of the current approach for the following use cases:
   containers should be able to use other free resources on the same
   device.
 
-  *Limitation*: Current implementation of the device plugin doesn’t
-  allow one to allocate part of the device because parameters are too limited
-  and Kubernetes doesn't have enough information about the extended
-  resources on a node to decide whether they can be shared.
+  *Limitation*: For example, newer generations of NVIDIA GPUs have a mode of
+  operation called MIG, that allow them to be sub-divided into a set of
+  mini-GPUs (called MIG devices) with varying amounts of memory and compute
+  resources provided by each. From a hardware-standpoint, configuring a GPU
+  into a set of MIG devices is highly-dynamic and creating a MIG device
+  tailored to the resource needs of a particular application is well
+  supported. However, with the current device plugin API, the only way to make
+  use of this feature is to pre-partition a GPU into a set of MIG devices and
+  advertise them to the kubelet in the same way a full / static GPU is
+  advertised. The user must then pick from this set of pre-partitioned MIG
+  devices instead of having one created for them on the fly based on their
+  particular resource constraints. Without the ability to create MIG devices
+  dynamically (i.e. at the time they are requested) the set of pre-defined MIG
+  devices must be carefully tuned to ensure that GPU resources do not go unused
+  because some of the pre-partioned devices are in low-demand.  It also puts
+  the burden on the user to pick a particular MIG device type, rather than
+  declaring the resource constraints more abstractly.
```

Comment:
Nit: Picking a MIG device type (a name for a pre-configured set of resources) would be the more-abstract option, I think what we want (and get) from this KEP is being able to declare our resource constraints more "explicitly".

Reply:
I guess it depends on how you look at it. By “more abstractly” I meant you might only care about making sure you get at least 5GB of memory and then the plugin can either create a MIG device that satisfies that for you, or give you a full GPU if Tafts all that’s available and it’s OK with that. With pre-defined MIG devices a user has to explicitly request that device type rather than providing a more abstract set of resource constraints.

Reply:
That's fair. I read the case we're trying to enable not as "choosing the best GPU", but subdividing a GPU to match your explicit requirements on-demand, vs choosing a pre-defined GPU *division* (I thought that's what "MIG device type" meant here) which probably matches your requirements, but is abstracted behind a name.

Reply:
True. Looking at just the "resource sharing" / subdivison aspect what you say makes sense. I ended up conflating 2 different advantages of the new proposal into 1, i.e. better support for "resource sharing" and better support for matching workloads to the "right sized" resources based on an abstract set of constraints.